tower/src/util/mod.rs

```diff
@@ -61,6 +61,18 @@ pub mod future {
 /// adapters
 pub trait ServiceExt<Request>: tower_service::Service<Request> {
     /// Yields a mutable reference to the service when it is ready to accept a request.
+    fn ready(&mut self) -> Ready<'_, Self, Request>
+    where
+        Self: Sized,
+    {
+        Ready::new(self)
+    }
+
+    /// Yields a mutable reference to the service when it is ready to accept a request.
+    #[deprecated(
+        since = "0.4.6",
+        note = "Please use the ServiceExt::ready method instead"
+    )]
     fn ready_and(&mut self) -> ReadyAnd<'_, Self, Request>
```

Comment:
we need to stick an `allow(deprecated)` somewhere so this doesn't warn due to the use of the `ReadyAnd` type.